### PR TITLE
enable HostPath PV provisioning by default

### PR DESCRIPTION
**Pull request by wackxu**
**What this PR does / why we need it**:
Since HostPath PV provisioning is only used for testing and development of provisioning features in single node, we should enable it by default to make test easy in local up environment.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by msau42** [comment]
Is local-up only used for testing purposes?  The only thing I'm concerned about is if anyone uses it for production. Then they will get very surprised they lose all their data and realize hostpath PV was in /tmp.

**Comment by dims** [comment]
@msau42 yes, only for testing.

**Comment by cblecker** [comment]
Yes, local-up should only be used for testing. That said, if there is a risk of data loss, even in testing, we should call it out with a warning.

**Comment by msau42** [comment]
We call it out in the docs and official tutorials that hostpath pv should only be used for testing purposes. But still, it gets missed. The user experience has been:
* Try out tutorial on local cluster
* PV provisioning fails
* User tries to figure out the error, eventually figures out that hostpath provisioner needs to be enabled, and hopefully gets explanation of why it should only be used for testing.

With this change, we run the risk of the user not being aware of the limitations because everything will succeed.

**Comment by wackxu** [comment]
@msau42  When I run local up, I found there is a default storageclas

```
NAME                 PROVISIONER               AGE
standard (default)   kubernetes.io/host-path   6m
```
I have know that the host_path provisioner is used only for development and testing only and WILL NOT WORK in a multi-node cluster.But when I create a pvc and refer to the default provisioner , the provision is failed because I do not enable the host path provisioner, so I need to find out how to enable it, It is not friendly  for users.

**Comment by msau42** [comment]
I agree it's not great that it works out of the box.  But I also don't want users to not be aware of the limitations and potentially end up storing production data on it.  I don't see a great way to be able to "warn" users that hostpath provisioner is only for dev/testing.  It's already in the docs, but it's very easy to miss if you don't read them.

**Comment by wackxu** [comment]
@msau42 But local-up is only  a single node cluster and only used for testing. I think the docs is enough to "warn" users that hostpath provisioner is only for dev/testing when users deploy product cluster.

**Comment by jsafrane** [comment]
My opinion is that we should probably enable the provisioner, however it would be nice to add a yellow (or even red) warning into `hack/local-up-cluster` output that the provisioner is just for testing.

File: hack/local-up-cluster.sh [hunk 1]
```diff
@@ -233,7 +233,7 @@ CONTAINER_RUNTIME_ENDPOINT=${CONTAINER_RUNTIME_ENDPOINT:-""}
 IMAGE_SERVICE_ENDPOINT=${IMAGE_SERVICE_ENDPOINT:-""}
 CHAOS_CHANCE=${CHAOS_CHANCE:-0.0}
 CPU_CFS_QUOTA=${CPU_CFS_QUOTA:-true}
-ENABLE_HOSTPATH_PROVISIONER=${ENABLE_HOSTPATH_PROVISIONER:-"false"}
```
Comment (jsafrane) [inline hack/local-up-cluster.sh]: You should enable it only when you create the default storage class for hostpath, i.e. somewhere in `create_storage_class`. The script deploys real AWS/GCE/OpenStack/... storage class in the clouds and it does not need hostpath provisioner there (unless user enables it explicitly via env. variable).

**Comment by wackxu** [comment]
@msau42 @jsafrane Updated, the warning message like below:

```
root@node1:/xsw/gowork/src/k8s.io/kubernetes# ./hack/local-up-cluster.sh 
WARNING : The kubelet is configured to not fail even if swap is enabled; production deployments should disable swap.
WARNING : The hostpath provisioner is used only for development and testing only and WILL NOT WORK in a multi-node cluster.
```

**Comment by jsafrane** [comment]
/lgtm cancel

Got second thoughts. What happens when user hits ^C? Looking at the code, it leaves the host path volumes and their data in `/tmp/hostpath_pv/` (hardcoded in host_path.go). Should there be some cleanup in local-up-cluster.sh?

**Comment by wackxu** [comment]
@jsafrane When users use ^C to force stop local cluster, I think just leave the data there and let users manually clear it because users maybe want to use the data for analyse and so on. Just like we manually enable the host path provisioner.

**Comment by jsafrane** [comment]
Hmm, I've never needed any data when I stopped my local cluster - I use local-up-cluster.sh to debug Kubernetes, not applications on top of it. I admin different people can use it for different purposes.

We can always add some HOSTPATH_PROVISIONER_CLEANUP variable later.

/lgtm

**Comment by jsafrane** [comment]
@cblecker, please look at the discussion right above and approve / disapprove.

File: hack/local-up-cluster.sh [hunk 2]
```diff
@@ -150,6 +151,11 @@ if [ "$(id -u)" != "0" ]; then
     echo "WARNING : This script MAY be run as root for docker socket / iptables functionality; if failures occur, retry as root." 2>&1
 fi
 
+if [ -z "$CLOUD_PROVIDER" ]; then
+    ENABLE_HOSTPATH_PROVISIONER=true
```
Comment (cblecker) [inline hack/local-up-cluster.sh]: The problem with this logic is there is no way to force disable the hostpath provisioner.

We should not set the default above, and then in this logic, check if it is unset as a condition. Then afterwards, set the false default if unset.

File: hack/local-up-cluster.sh [hunk 3]
```diff
@@ -150,6 +151,11 @@ if [ "$(id -u)" != "0" ]; then
...
+    echo "WARNING : The hostpath provisioner is used only for development and testing only and WILL NOT WORK in a multi-node cluster."
```
Comment (cblecker) [inline hack/local-up-cluster.sh]: This indicates that the hostpath provision is used for development, but doesn't actually say that the script is taking the action of enabling it.

**Review by cblecker — CHANGES_REQUESTED** [review]

**Pull request closed without merging**